component/loki/process/internal/stages/pipeline.go

```diff
@@ -20,8 +20,11 @@ import (
 // We define these as pointers types so we can use reflection to check that
 // exactly one is set.
 type StageConfig struct {
```

Comment:
Hm, I just noticed our component is using this directly in its arguments, which means that a publicly exposed type has a public field of an internal type, preventing external callers from being able to construct arguments by hand. 

I'm not sure if it's worth fixing right now, but maybe we should have a comment here to resolve in the future?

Reply:
Yeah that's true. I'm adding a TODO comment, would the proper solution be to make the package non-internal once we've gained some confidence?